target/linux/ath79/image/generic.mk

```diff
@@ -314,6 +331,27 @@ define Device/avm_fritzdvbc
 endef
 TARGET_DEVICES += avm_fritzdvbc
 
+define Device/belkin_f9j1108-v2
+  SOC := qca9558
+  DEVICE_VENDOR := Belkin
+  DEVICE_MODEL := F9J1108v2 (AC1750 DB Wi-Fi)
+  DEVICE_PACKAGES += kmod-kmod-ath10k-ct ath10k-firmware-qca988x-ct kmod-usb2 kmod-usb3 kmod-usb-ledtrig-usbport
+  KERNEL_SIZE := 5120k
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x50000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | lzma | \
+	uImage lzma
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  IMAGE/sysupgrade.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | \
+	pad-rootfs | append-metadata | check-size
+  IMAGE/factory.bin := edimax-headers $(1) F9J1108v1 BR-6679BAC | append-rootfs | append-kernel | check-size
```

Comment:
`append-rootfs | append-kernel`

do you mean

`append-kernel | append-rootfs`
?

we want the loader to be before the actual kernel correct?

Reply:
The factory bin is actually dissected by u-boot, and installed automatically into the MTD defined in the u-boot configuration.  The header uImage and rootfs tells u-boot what data to put where.
When I tested kernel -> rootfs order, it only installed the kernel really quickly and skipped over the rootfs.

Reply:
fast time for kernel is expected because its the loader only, but it skipped the rest really? wow... any errors? any idea why?

Reply:
what if there was some padding in between?

Reply:
I tried first with padding, then without padding, then in the reverse order.  I will try again now and capture logs to see if my brain was playing tricks on me

Reply:
OK, it must have been a build error that caused it last time, reflashing just then worked with uimage first:

Re-tune image start to 810002fc, len to 5640423
Bytes transferred = 5640423 (5610e7 hex)
http upload ready
http ugrade ...
######################### Eraseall 0 ########################
checksum = e4
checksum = 16
Header Data:
Model Name:     BR-6679BAC
Data Size:      3329 Bytes
Start Address:  0x9FE70000
End Address:    0x9FE70D00
MTD Name:       uImage
Version:        OpenWrtr14365
protect off all
Un-Protect Flash Bank # 1
erase 9fe70000 +d01
Erasing flash...
First 0xe7 last 0xe7 sector size 0x10000
 231
Erased 1 sectors
cp.b 8100035a 9fe70000 d01
Copy to Flash... write addr: 9fe70000
done
######################### Eraseall 1 ########################
checksum = 9b
checksum = 21
Header Data:
Model Name:     BR-6679BAC
Data Size:      5636906 Bytes
Start Address:  0x9F050000
End Address:    0x9F5B0329
MTD Name:       rootfs
Version:        OpenWrtr14365
protect off all
Un-Protect Flash Bank # 1
erase 9f050000 +56032a
Erasing flash...
First 0x5 last 0x5b sector size 0x10000
  91
Erased 87 sectors
cp.b 810010b9 9f050000 56032a
Copy to Flash... write addr: 9f050000
done
######################### Eraseall 2 ########################
run_cmd error!!
#############################################################
reset